### PR TITLE
Update to VS2019.

### PR DESCRIPTION
**Pull request by AWoloszyn**
This involves updating llvm and fixing some bugs that this
introduced.

### CODE REVIEW

File: Libraries/WNScripting/CMakeLists.txt [hunk 1]
```diff
@@ -156,6 +156,15 @@ if (NOT WN_DISABLE_CXX)
         COMMAND ${CMAKE_COMMAND} --build . ${WN_LLVM_ADDITIONAL_BUILD_FLAGS} --target ${WNScriptingLLVMLibs}
       WORKING_DIRECTORY ${WNFramework_BINARY_DIR}/Externals/llvm-project/llvm)
 
+  set_target_properties(
```
Comment (devpow112) [inline Libraries/WNScripting/CMakeLists.txt:159]: Should this go into the multibuild overlay or windows overlay?

File: Overlays/Windows/Overlays/D3D12/Runtime/WNGraphics/src/D3D12/WNDevice.cpp [hunk 2]
```diff
@@ -40,6 +40,7 @@
 
 #include <d3d12.h>
 #include <dxgi1_4.h>
+#include <functional>
```
Comment (devpow112) [inline Overlays/Windows/Overlays/D3D12/Runtime/WNGraphics/src/D3D12/WNDevice.cpp:43]: What do we use from this header in here?

File: cmake/targets.cmake [hunk 3]
```diff
@@ -447,6 +447,7 @@ function(wn_filesystem_files name)
     add_custom_target(${name} DEPENDS 
        ${CMAKE_CURRENT_BINARY_DIR}/${PARSED_ARGS_OUTPUT_DIR}/${name}.h
        ${CMAKE_CURRENT_BINARY_DIR}/${PARSED_ARGS_OUTPUT_DIR}/${name}.cpp)
+    overlay_named_file(cmake/target_functions/post_filesystem_files.cmake)
```
Comment (devpow112) [inline cmake/targets.cmake:450]: WHat does this add exactly?

File: Libraries/WNScripting/CMakeLists.txt (hunk 1 above)
Comment (AWoloszyn) [reply Libraries/WNScripting/CMakeLists.txt:159]: Probably, I wasnt sure where to put it

File: Overlays/Windows/Overlays/D3D12/Runtime/WNGraphics/src/D3D12/WNDevice.cpp (hunk 2 above)
Comment (AWoloszyn) [reply Overlays/Windows/Overlays/D3D12/Runtime/WNGraphics/src/D3D12/WNDevice.cpp:43]: bind comes from here. (2019 it doesnt get transitively included)

File: cmake/targets.cmake (hunk 3 above)
Comment (AWoloszyn) [reply cmake/targets.cmake:450]: This lets us put the wn_filesystem_files into overlay directories.

**Review by devpow112 — DISMISSED** [review]
Looks good to me

**Review by devpow112 — APPROVED** [review]

**Pull request merged**